lib/Util/Security.php

```diff
@@ -70,7 +71,7 @@ static public function generateUnsubscribeToken($model) {
 
   public function generateUnsubscribeTokenByEntity($entity): string {
     $repository = null;
-    if ($entity instanceof NewsletterEntity) {
+    if ($entity instanceof NewsletterEntity || $entity instanceof SubscriberEntity) {
```

Comment:
Hmm, I don't think this is going to work. If the instance is `SubscriberEntity` we still use the newsletters repository and line 82 down bellow would never work.

Reply:
Yes, you are right. This is my stupid mistake. Thank you for the catch.